sdk/core/src/services/syft/mod.rs

```diff
@@ -231,38 +217,26 @@ impl Service {
             env::set_current_dir(absolute_path).map_err(Error::Io)?
         }
 
-        let output = match cataloger.clone() {
-            Some(cataloger) => self.run_syft_with_cataloger(cataloger),
-            None => self.run_syft(),
-        }?;
+        let output = self.run_syft(cataloger.clone())?;
 
         // Handle error generated by syft.
         if !&output.status.success() {
-            let error_msg = String::from_utf8(output.stderr)
-                .map_err(Error::Utf8Conversion)?;
+            let error_msg = String::from_utf8(output.stderr).map_err(Error::Utf8Conversion)?;
             return Err(Error::Syft(error_msg));
         }
 
         if output.stdout.is_empty() {
             return Err(Error::Syft("syft generated empty SBOM".to_string()));
         };
 
-        let output: String = String::from_utf8_lossy(&output.stdout)
-            .to_string();
+        let output: String = String::from_utf8_lossy(&output.stdout).to_string();
 
-        let json: Value = serde_json::from_str(&output)
-            .map_err(Error::Serde)?;
+        let json: Value = serde_json::from_str(&output).map_err(Error::Serde)?;
 
-        let mut sbom: Bom = serde_json::from_value(json)
-            .map_err(Error::Serde)?;
+        let mut sbom: Bom = serde_json::from_value(json).map_err(Error::Serde)?;
```

Comment:
Did you know we have `Bom::parse` to parse Boms? I ask because I've run into various error conditions, and in the Ion Channel branch I'm already having to extend that to handle NDJson which will fail.  It seems like using a common method everywhere might be smart to get consistent or unpredictable edge cases handled.

Reply:
No, I had no idea.  It's not in the `sdk/core/src/entities/cyclonedx/models/bom.rs` file and the file was generated so I assumed that it didn't have any static functions.  Where are we defining it?  I'll look

Reply:
OMG, that is really cool feature of Rust: The ability to generate types and add implementation elsewhere.  I wonder how I can tell if a generated type has implementation somewhere?  I'd have never thought to look